Python/search/dfs.py

```diff
@@ -1,58 +1,165 @@
-            for i in self.graph[s]: 
-                if visited[i] == False: 
-                    queue.append(i) 
-                    visited[i] = True
-  
-# Driver code 
+# Depth First Search Traversal in Graphs
+
+# Graph class created
+class Graph:
+
+    def __init__(self):
+        # graph declared to a dictionary
+        self.graph = {}
+        # visited nodes are maintained in this set
+        self.visited = set()
+
+    def add_vertex(self, vertex):
+        # vertex is added to the graph dictionary and it's adjacency list is declared
+        self.graph[vertex] = []
+    
+    def add_edge(self, source, destination):
+        # directional edges are added in this function
+        # the edge is directional, i.e., going from source to destination
+
+        # conditional statements in case the vertex aren't present in the graph
+        if destination not in self.graph or source not in self.graph:
+            if source not in self.graph:
+                print(str(source)+' is not a vertex')    
+                # 1 is returned to avoid stopping the running of the program allows the user to re-enter values
+                return 1 
+            if destination not in self.graph:
+                print('\n'+str(destination)+' is not a vertex.')
+                # 1 is returned to avoid stopping the running of the program allows the user to re-enter values
+                return 1     
+
+        # edge is added to the graph, from source to destination
+        self.graph[source].append(destination)
+
+    def add_edge_bidirected(self, vertex1, vertex2):
+        # bidirectional graph edges are added in this function
+        # the edge is bidirectional, i.e., the edge goes both ways
+
+        # conditional statements to check if the vertices are present
+        if vertex2 not in self.graph or vertex1 not in self.graph:
+            if vertex1 not in self.graph:
+                print('\n'+str(vertex1)+' is not a vertex'+'\n')
+                return 1    # 1 is returned to avoid stopping the running of the program allows the user to re-enter values
+            if destination not in self.graph:
+                print('\n'+str(vertex2)+' is not a vertex.'+'\n')
+                return 1    # 1 is returned to avoid stopping the running of the program allows the user to re-enter values
+        
+        # edges are added to the adjacency list
+        self.graph[vertex1].append(vertex2)
+        self.graph[vertex2].append(vertex1)
+
+    def dfs_helper(self, start):
+        # function to assist in calling dfs
+        # also prints the final dfs traversal
+
+        # checks if the start node is in the graph
+        if start not in self.graph:
+            print('\n'+str(start)+' not in graph.'+'\n')
+            # 1 is returned to avoid stopping the running of the program,
+            # allows the user to re-enter the start vertex
+            return 1 
```

Comment:
Instead of returning 1 when there is an exception you can throw exceptions

Reply:
Also, it looks you are putting in a lot of efforts and it's always good to see such enthusiastic contributors and work with them.

Reply:
Thank you for your guidance. I have learnt quite a lot with just this one PR. Thank you so much for that.